lang/java/reef-bridge-java/src/main/java/org/apache/reef/javabridge/FailedTaskBridge.java

```diff
@@ -46,29 +54,60 @@ public FailedTaskBridge(final FailedTask failedTask, final ActiveContextBridgeFa
     } else {
       this.jactiveContext = null;
     }
+
+    try {
+      this.failedTaskSerializedAvro = generateFailedTaskSerializedAvro();
+    } catch(final Exception e) {
+      throw new RuntimeException(e);
+    }
   }
 
   public ActiveContextBridge getActiveContext() {
     return jactiveContext;
   }
 
-  public String getFailedTaskString() {
-    final String description = jfailedTask.getDescription().isPresent() ?
-        jfailedTask.getDescription().get().replace("=", "").replace(",", "") : "";
-    final String cause = jfailedTask.getReason().isPresent() ?
-        jfailedTask.getReason().get().toString().replace("=", "").replace(",", "") : "";
-    final String data = jfailedTask.getData().isPresent() ?
-        new String(jfailedTask.getData().get(), StandardCharsets.UTF_8).replace("=", "").replace(",", "") : "";
+  public byte[] getFailedTaskSerializedAvro() {
+    return failedTaskSerializedAvro;
+  }
+
+  private byte[] generateFailedTaskSerializedAvro() throws IOException {
+    AvroFailedTask avroFailedTask = null;
+
+    if (jfailedTask.getData() != null && jfailedTask.getData().isPresent()) {
+      // Deserialize what was passed in from C#.
+      try (final ByteArrayInputStream fileInputStream = new ByteArrayInputStream(jfailedTask.getData().get())) {
+        final JsonDecoder decoder = DecoderFactory.get().jsonDecoder(
```

Comment:
This is a one-time thing. I don't see the need of creating a special class for it.